libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -233,14 +233,18 @@ pair<bool, shared_ptr<SymbolicVariable>> newSymbolicVariable(
 	bool abstract = false;
 	shared_ptr<SymbolicVariable> var;
 	frontend::Type const* type = &_type;
-	if (!isSupportedTypeDeclaration(_type))
+
+	if (auto userType = dynamic_cast<UserDefinedValueType const*>(type))
+		type = &userType->underlyingType();
```

Comment:
```suggestion
	if (auto userType = dynamic_cast<UserDefinedValueType const*>(type))
		return newSymbolicVariable(&userType->underlyingType(), _uniqueName, _context);
```
This is what I meant. The rest of the code can remain unchanged.

Reply:
Ah I see

Reply:
I left the other changes in this file just cus it looked better